server/src/test/java/org/elasticsearch/cluster/routing/allocation/decider/DiskThresholdDeciderUnitTests.java

```diff
@@ -901,6 +904,74 @@ public void testDecidesYesIfWatermarksIgnored() {
         assertThat(decision.getExplanation(), containsString("disk watermarks are ignored on this index"));
     }
 
+    public void testDecidesYesIfPromotableShardRole() {
+        final Metadata metadata = Metadata.builder()
+            .put(IndexMetadata.builder("test").settings(settings(IndexVersion.V_8_10_0)).numberOfShards(1).numberOfReplicas(0))
+            .build();
+        final Index index = metadata.index("test").getIndex();
+        ShardRouting shardRouting = ShardRouting.newUnassigned(
+            new ShardId(index, 0),
+            true,
+            EmptyStoreRecoverySource.INSTANCE,
+            new UnassignedInfo(UnassignedInfo.Reason.INDEX_CREATED, "foo"),
+            ShardRouting.Role.INDEX_ONLY
+        );
+
+        final var node = DiscoveryNodeUtils.builder("node_id").name("node_name").roles(Set.of(DiscoveryNodeRole.INDEX_ROLE)).build();
+        ClusterState clusterState = ClusterState.builder(ClusterName.DEFAULT)
+            .metadata(metadata)
+            .routingTable(RoutingTable.builder(new ShardRoutingRoleStrategy() {
+                @Override
+                public ShardRouting.Role newEmptyRole(int copyIndex) {
+                    return ShardRouting.Role.INDEX_ONLY;
+                }
+
+                @Override
+                public ShardRouting.Role newReplicaRole() {
+                    return ShardRouting.Role.DEFAULT;
```

Comment:
Why not `SEARCH_ONLY`?

Reply:
The change only impacts shards with INDEX_ONLY roles and does not use replicas.

Reply:
If we have no replicas could we assert that this method isn't called instead?

Reply:
Sure, I pushed 52b0c45b6330ebd41fdb2b5e536e8f94cd0068ee